### DISCUSSION TITLE
PostgreSQL pipelining removed from the Techempower benchmark

### DISCUSSION THREAD

**Discussion by selforganized** [matt-42/lithium#116] [General]
Hi,

I saw that you've removed PostgreSQL pipelining from the Techempower benchmark ([#7328](https://github.com/TechEmpower/FrameworkBenchmarks/pull/7382)). The reason given was to comply with the benchmark rules. What rules did Lithium break? Other frameworks are still using PG pipelining.

Thanks.

**Comment by matt-42** [comment]
It has been forbidden after the TFB team stated this as an unsafe optimization, even for read only usecases. They asked all frameworks to remove pipelining support for the round 21: https://github.com/TechEmpower/FrameworkBenchmarks/issues/7289#issue-1200206687

**Comment by selforganized** [comment]
I see. I noticed that Lithium's Fortune performance dropped from 670,000/s to 380,000/s without pipelining, an almost 45% decrease.

But the performances of other top-ranking frameworks, such as drogon, actix-http, and just-js, saw no decrease even after removing pipelining. Do you know why the removal of pipelining affects Lithium so much more than those frameworks? Thanks.

**Comment by matt-42** [comment]
I do not know why. I just had time to deactivate pipelining to avoid having lithium being ejected from the benchmark but I did not have time to investigate more. 
This may be due to lithium batching pgsql request across several http requests together which gives a bigger speedup than just batching the pgsql request of one http resquest, especially for http requests like fortunes triggering only one pgsql query.
